packages/server-wallet/src/e2e-testing/wallet-load-node.ts

```diff
@@ -0,0 +1,274 @@
+        .forEach(s =>
+          setTimeout(async () => {
+            this.logger.trace({step: s}, 'Starting job');
+
+            const result = await this.handleStep(s);
+            if (result.type !== 'Success') {
+              this.logger.error({result, step: s}, 'Load step failed!');
+              console.error(chalk.redBright(`Step failed for ${s.jobId}-${s.timestamp}`));
+              process.exit(1);
+            } else {
+              // We use a simple counter to determine when we're done
+
+              this.completedSteps++;
+              this.logger.trace({step: s, timestamp: Date.now() - startTimestamp}, 'Completed job');
+
+              const ourSteps = this.steps.filter(s => s.serverId === this.loadNodeConfig.serverId);
+              // We only care if we've completed our steps
+              if (this.completedSteps === ourSteps.length) {
+                this.logger.trace(
+                  {jobIds: this.jobToChannelMap.map(e => e.jobId)},
+                  'All jobs completed!'
+                );
+                console.log(chalk.greenBright(`All jobs complete!`));
+                resolve();
+              }
+            }
+          }, s.timestamp)
+        );
+    });
+  }
+
+  private async shareChannelIdsWithPeers(jobAndChannel: {channelId: string; jobId: string}) {
+    for (const {loadServerPort} of this.peers) {
+      await got.post(`http://localhost:${loadServerPort}/channelId/`, {json: jobAndChannel});
+    }
+  }
+  private async shareJobsWithPeers(steps: Step[]) {
+    for (const {loadServerPort} of this.peers) {
+      await got.post(`http://localhost:${loadServerPort}/load/?fromPeer=true`, {json: steps});
+    }
+  }
+  private async sendGetRequestToPeers(urlPath: string) {
+    for (const {loadServerPort} of this.peers) {
+      await got.get(`http://localhost:${loadServerPort}${urlPath}`, {retry: 0});
+    }
+  }
+
+  private async updateJobQueue(steps: Step[]): Promise<void> {
+    this.steps = _.uniq(this.steps.concat(steps));
+    console.log(chalk.yellow(`Updated job queue with ${steps.length} steps`));
+    this.logger.info({steps}, 'Updated queue with steps');
+  }
+
+  private getChannelIdForJob(jobId: string): string {
+    const entry = this.jobToChannelMap.find(e => e.jobId === jobId);
+    if (!entry) {
+      throw new Error(`No channel id for ${jobId}`);
+    }
+    return entry.channelId;
+  }
+
+  /**
+   * These are the various handlers for different step types
+   * TODO: Typing should not use any type for the request
+   */
+  private stepHandlers: Record<Step['type'], (req: any) => Promise<ObjectiveDoneResult>> = {
+    CreateChannel: async (request: CreateChannelStep) => {
+      const [result] = await this.serverWallet.createChannels([request.channelParams]);
+      const {jobId} = request;
+      const {channelId} = result;
+
+      this.jobToChannelMap.push({jobId, channelId});
+      await this.shareChannelIdsWithPeers({jobId, channelId});
```

Comment:
This seems to be the buggiest part of the system. An alternative is to embed the `jobId` in the app data of the channel. That way, the joiner has the opportunity to record the `jobId <-> channelId` mapping before approving the `joinChannel`. Is that too clever?

Reply:
> This seems to be the buggiest part of the system. An alternative is to embed the `jobId` in the app data of the channel. That way, the joiner has the opportunity to record the `jobId <-> channelId` mapping before approving the `joinChannel`. Is that too clever?

That's smart! Would there be any issue with not using "valid" `appData`? One possible scenario where we could run into trouble would be enabling EVM validation. Maybe challenging later on?

Reply:
When EVM validation is enabled, we can use rules where that app data is valid!

Reply:
That's a good point! 

Locally I haven't been running into the jobId issue, so I'm a bit hesitant to make that change yet. If we can avoid having to make channels "special" in any way for load testing I think that would be nice.

I'm going to revisit this after I make some of the load-generator changes you suggested. I'm hoping this is an artifact of having really small time values.